### ISSUE TITLE
Save state of lr_scheduler for additional training runs

### ISSUE THREAD

**Issue by ChrKoenig** [citoverse/cito#95]
It seems that only the initial state of the learning rate scheduler is saved in the model object under `model$training_properties$lr_scheduler`. The problem is that this initial state is also applied when running `continue_training` on a model that has not yet fully converged, which leads to an immediate spike in loss and subsequent shutdown with the message:

```Cancel training because loss is still above baseline, please hyperparameters. See vignette('B-Training_neural_networks') for help```

This effectively prohibits further training of any model that has undergone adjustments in its learning rate. 

The same issue might apply to the state of the optimizer.

**Comment by MaximilianPi** [comment]
Yes, this is a problem, I think initially we didn't save the state of the optimizer because it was too hard to save the state as a .RDS, however, with torch_serialize it shouldn't be a Problem @ArminSchenk
